src/pod.ts

```diff
@@ -387,6 +469,46 @@ export class Pod extends Resource implements IPodSpec {
 
 }
 
+/**
+ * Holds pod-level security attributes and common container settings.
+ */
+export class PodSecurityContext {
+
+  public readonly ensureNonRoot: boolean;
+  public readonly user?: number;
+  public readonly group?: number;
+  public readonly fsGroup?: number;
+  public readonly fsGroupChangePolicy: FsGroupChangePolicy;
```

Comment:
Are `ensureNonRoot` and `fsGroupChangePolicy` required here because they are required in the kubernetes manifest? Or because we are suggesting them as opinionated defaults for all pods / pod templates?

I imagine an alternative could be making all security fields optional, so that if no PodSecurityContext is provided to `PodSpecProps`, then the pod `securityContext` doesn't get rendered.

If we want to keep these defaults, perhaps you can add a note in the PR description to just explain why these defaults were chosen (since I think it's fair to say "sane defaults" is one the main features of cdk8s+. I imagine we have design tenets about this somewhere 😛).